### PR TITLE
Set RW mutex on config

### PR DESCRIPTION
**Pull request by jotak**

### CODE REVIEW

**Comment by jotak** [comment]
@jmazzitelli @lucasponce so i've also added a test that would illustrate the race when mutex is absent, when running `make test-race`

**Comment by rhqci** [comment]
Jenkins CI: kiali-core-pr-e2e-test [#829](https://paste.fedoraproject.org/paste/drydQDnCcdFBALJOiyseiw)
  * :heavy_check_mark: run-kiali-e2e-tests [#[1653]](https://paste.fedoraproject.org/paste/jGLNCPGQadJ01Kp2Mb3U4A)

File: config/config.go [hunk 1]
```diff
@@ -285,12 +287,17 @@ func NewConfig() (c *Config) {
...
+	rwMutex.RUnlock()
+	return &copy
 }
 
 // Set the global Config
 func Set(conf *Config) {
-	configuration = conf
+	rwMutex.Lock()
+	configuration = *conf
+	rwMutex.Unlock()
```
Comment (jmazzitelli) [inline config/config.go]: same suggestion as before:

```
	rwMutex.Lock()
	defer rwMutex.Unlock()
	configuration = *conf
```

**Comment by jotak** [comment]
@jmazzitelli @lucasponce update done

**Comment by rhqci** [comment]
Jenkins CI: kiali-core-pr-e2e-test [#836](https://paste.fedoraproject.org/paste/L8zrdXb6hMfvoKi5QbExnQ)
  * :heavy_check_mark: run-kiali-e2e-tests [#[1658]](https://paste.fedoraproject.org/paste/ZfujgBiaaG1VLlJ7UvQ1YQ)

**Pull request merged**